File: chess/Bishop.py
from .ChessPiece import ChessPiece

class Bishop(ChessPiece):
    def __init__(self, player, pos):
        super().__init__(player, pos)
        self.multipleMove = True
        self.canJumpOverOthers = False
        self.possibleRegularMoves = self.possibleCaptureMoves = [(1, 1), (1, -1), (-1, 1), (-1, -1)]
        self.name = 'Bishop'

File: chess++/Bishop.cpp
#include "Bishop.h"

Bishop::Bishop(const Player& player, const Position& position)
: ChessPiece(player, position) {
    this->name = "Bishop";
    this->multipleMove = true;
    this->canJumpOverOthers = false;

    // Initializing possible moves for a bishop
    this->possibleRegularMoves = {
        {1, 1}, {1, -1}, {-1, 1}, {-1, -1}
    };
    // For Bishop, capture moves are the same as regular moves
    this->possibleCaptureMoves = this->possibleRegularMoves;
}
